Cross-compile libsavi_runtime.bc for DragonFly

mneumann opened the pull request:
On my FreeBSD system, I can cross-compile the `libsavi_runtime.bc` for DragonFly as follows:
```
gmake -f Makefile.dragonfly
```

Note that this depends on this patch (not needed when integrated into the .cirrus.yml):

```
diff --git a/patches/non-relative-llvm-link.patch b/patches/non-relative-llvm-link.patch
index c9e701d..25354cd 100644
--- a/patches/non-relative-llvm-link.patch
+++ b/patches/non-relative-llvm-link.patch
@@ -7,7 +7,7 @@ index 06ef5919..7ca37f50 100644
  
      add_custom_target(libponyrt_bc ALL
 -        ${libponyrt_SOURCE_DIR}/../../build/libs/bin/llvm-link -o "${libponyrt_BINARY_DIR}/libponyrt.bc" ${_ll_bc_all}
-+        llvm-link -o "${libponyrt_BINARY_DIR}/libponyrt.bc" ${_ll_bc_all}
++        ${LLVM_LINK} -o "${libponyrt_BINARY_DIR}/libponyrt.bc" ${_ll_bc_all}
          DEPENDS ${_ll_bc_all}
      )
```


@jemc: Maybe you can add this to the `.cirrus.yml` file or guide me a bit. This uses a DragonFly sysroot provided by Ravenports (www.ravenports.com / a cross-platform ports system).

mneumann committed:
Cross-compile libsavi_runtime.bc from FreeBSD

```
gmake -f Makefile.dragonfly
```

jemc commented:
Where did the `execinfo.h` and `stdatomic.h` files come from?

Ideally I'd prefer to use `curl`/`wget`/`git` to fetch them from some remote source rather than needing to include them here in the repo.

Other than needing that bit of info, yes, I can take a shot at adapting the `.cirrus.yml` code based on the example Makefile you gave.

mneumann commented:
`execinfo.h` comes from a DragonFly installation. `stdatomic.h` from FreeBSD + minor patch. Let me figure out downloads tomorrow.

mneumann commented:
https://github.com/DragonFlyBSD/DragonFlyBSD/blob/c163a4d7ee9c6857ee4e04a3a2cbb50c3de29da1/lib/libexecinfo/execinfo.h

https://github.com/freebsd/freebsd-src/blob/c1a24b9dbeade11bfeedc91996672582f6ea32f2/sys/sys/stdatomic.h

Plus this patch:

```
--- stdatomic.h 2022-09-30 00:13:23.344421000 +0200
+++ files/dfly/stdatomic.h      2022-09-29 20:26:37.276346000 +0200
@@ -33,7 +33,11 @@
 #define        _STDATOMIC_H_

 #include <sys/cdefs.h>
+#if defined(__FreeBSD__)
 #include <sys/_types.h>
+#elif defined(__DragonFly__)
+#include <sys/types.h>
+#endif

 #if __has_extension(c_atomic) || __has_extension(cxx_atomic)
 #define        __CLANG_ATOMICS
@@ -163,11 +167,6 @@
 #endif
 }

-#if defined(__cplusplus) && !defined(_Bool)
-#define        _Bool   bool
-#define        __bool_locally_defined
-#endif
-
 /*
  * 7.17.5 Lock-free property.
  */
@@ -200,8 +199,10 @@
 typedef _Atomic(unsigned long)         atomic_ulong;
 typedef _Atomic(long long)             atomic_llong;
 typedef _Atomic(unsigned long long)    atomic_ullong;
+#if !defined(__DragonFly__)
 typedef _Atomic(__char16_t)            atomic_char16_t;
 typedef _Atomic(__char32_t)            atomic_char32_t;
+#endif
 typedef _Atomic(___wchar_t)            atomic_wchar_t;
 typedef _Atomic(__int_least8_t)                atomic_int_least8_t;
 typedef _Atomic(__uint_least8_t)       atomic_uint_least8_t;
@@ -410,10 +411,5 @@
        atomic_flag_clear_explicit(__object, memory_order_seq_cst);
 }
 #endif /* !_KERNEL */
-
-#ifdef __bool_locally_defined
-#undef _Bool
-#undef __bool_locally_defined
-#endif

 #endif /* !_STDATOMIC_H_ */
```

jemc committed:
Add `x86_64-unknown-dragonfly` to CI platform matrix.

jemc committed:
Remove old files related to DragonFly compilation.

These are no longer needed now that the build is happening
in `.cirrus.yml` without a reliance on these files.
